src/ui/components/map/providers/googlemapprovider.js

```diff
@@ -23,6 +23,35 @@ export default class GoogleMapProvider extends MapProvider {
     if (!this.hasValidClientCredentials() && !this._apiKey) {
       throw new Error('GoogleMapsProvider: Missing `apiKey` or {`clientId`, `signature`}');
     }
+
+    /**
+     * Language of the map.
+     * @type {string}
+     */
+    this._language = this.getSupportedLanguage(this._locale);
+  }
+
+  /**
+   * Google Maps supports some langauge codes that are longer than two characters. If the
+   * locale matches one of these edge cases, use it. Otherwise, fallback on the first two
+   * characters of the locale and print a warning.
+   * @param {string} localeStr Unicode locale
+   */
+  getSupportedLanguage (localeStr) {
+    const googleMapsCustomLanguages =
+      ['zh-CN', 'zn-HK', 'zh-TW', 'en-AU', 'en-GB', 'fr-CA', 'pt-BR', 'pt-PT', 'es-419'];
+    const locale = localeStr.replace('_', '-');
+    const language = locale.substring(0, 2);
```

Comment:
Small thing, but can we move `const language = ...` into the final `else-if` case? Having it here reads a little strangely. Seeing it above all the conditionals, I expect that the different cases are going to process `language` in some way to produce the correct answer. But, it's just used in the one conditional branch.

Reply:
Sure, that makes sense